src/utilities/apiImpl.ts

```diff
@@ -0,0 +1,51 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+import * as sarifApi from "./../api/sarifViewerApi";
+import * as vscode from "vscode";
+import { LogReader } from "../logReader";
+import { SVDiagnosticCollection } from "../svDiagnosticCollection";
+import { openSarifFile } from "../extension";
+
+export class ApiImpl implements sarifApi.Api {
+    /**
+     * Constructs an instance of the API given out by the extension.
+     * @param logReader An instance of the log reader to use for parsing log files.
+     * @param diagnosticCollection The diagnostic collection that holds the results.
+     */
+    public constructor(private readonly logReader: LogReader, private readonly diagnosticCollection: SVDiagnosticCollection) {
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public async openLogFiles(sarifFiles: vscode.Uri[], openLogFileArguments?: sarifApi.OpenLogFileArguments): Promise<void> {
+        for (const sarifUri of sarifFiles) {
+            await openSarifFile(sarifUri, this.logReader, this.diagnosticCollection, {
+                closeOriginalFileOnUpgrade: true,
+                openInTextEditor: openLogFileArguments?.openInTextEditor === undefined ? true : openLogFileArguments.openInTextEditor,
+                promptUserForUpgrade: openLogFileArguments?.promptUserForUpgrade === undefined ? true : openLogFileArguments.promptUserForUpgrade
+            });
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public async closeLogFiles(sarifFiles: vscode.Uri[]): Promise<void> {
+        for (const sarifUri of sarifFiles) {
+            this.diagnosticCollection.removeRuns(sarifUri);
+        }
+
+        return Promise.resolve();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public async closeAllLogFiles(): Promise<void> {
+        this.diagnosticCollection.removeAllRuns();
+        return Promise.resolve();
```

Comment:
Is your convention to return `Promise.resolve()` rather than just `undefined` (which the `async` will wrap for you)? #pattern

On a non-void async function, would you likewise return `Promise.resolve(whateverValue)`?

Reply:
Ah.. This function doesn't have an await, but I wanted to leave "Wiggle room" in the API  for it to be async.

If you just do public closeLogFiles(): Promise<void> with no async, the linter complains.. Hmm.. let me look into this.

---
In reply to: [409121885](https://github.com/microsoft/sarif-vscode-extension/pull/248#discussion_r409121885) [](ancestors = 409121885)

Reply:
I mean... keep the `async` modifier. But just change the return value.